xds/internal/server/listener_wrapper.go

```diff
@@ -297,11 +302,20 @@ func (l *listenerWrapper) Accept() (net.Conn, error) {
 			conn.Close()
 			continue
 		}
-		// TODO: once matched an accepted connection to a filter chain,
-		// instantiate the HTTP filters in the filter chain + the filter
-		// overrides, pipe filters and route into connection, which will
-		// eventually be passed to xdsUnary/Stream interceptors.
-		return &connWrapper{Conn: conn, filterChain: fc, parent: l}, nil
+		var rc xdsclient.RouteConfigUpdate
+		if fc.InlineRouteConfig != nil {
+			rc = *fc.InlineRouteConfig
+		} else {
+			l.rdsMu.RLock()
+			rc = l.rdsUpdates[fc.RouteConfigName]
+			l.rdsMu.RUnlock()
+		}
+
+		if err := fc.ConstructUsableRouteConfiguration(rc); err != nil {
+			// TODO: is this correct? Will this invoke the right behavior in GRPCServer?
```

Comment:
What needs to be done to remove this?

Reply:
I don't see any way to remove this error being raised. At the very least, converting a route to a matcher has potential to raise errors, similar to NewConfigSelector.

Reply:
I meant: "what needs to be done to remove the TODO?"

Reply:
Oh I switched it to what I believe to be correct, and thus deleted the TODO.